### PR TITLE
Update uninstall.sh

### PR DESCRIPTION
**Pull request by Gnouc**
- Need dot `.` for POSIX compliant instead of `source`
 - Use modern syntax for command substitution
 - Fix missing double quotes

### CODE REVIEW

File: tools/uninstall.sh [hunk 1]
```diff
@@ -18,14 +18,14 @@ then
 
   if [ -f ~/.zshrc ] || [ -h ~/.zshrc ]
   then
```
Comment (sachin21) [inline tools/uninstall.sh]: Can you change this style?
```
if [ -f ~/.zshrc ] || [ -h ~/.zshrc ]; then
```

ref: https://github.com/robbyrussell/oh-my-zsh/wiki/Coding-style-guide

Comment (Gnouc) [inline tools/uninstall.sh]: @sachin21 well, updated it.

Comment (sachin21) [inline tools/uninstall.sh]: :+1: :+1: :+1: :+1: :+1:

**Comment by Gnouc** [comment]
@robbyrussell

**Comment by ncanceill** [comment]
HI, and thanks for contributing.

I agree on everything. Keep in mind that `.` versus `source` is not an easy debate to settle.

On the one hand `.` is indeed the POSIX standard, and some shells do not support `source`. On the other hand, some other shells (mainly C-shells) do not support `.`.

Since the uninstall script should only be used through the `uninstall_oh_my_zsh` function which explicitly uses `sh`, it should work OK. However if someone tried to run it with `tcsh` it would fail. Maybe we should add a shebang?

@mcornella an opinion?

**Comment by Gnouc** [comment]
@ncanceill: I think the shebang didn't have and the best thing we can do is add the instruction for user to use a POSIX shell.

In case of csh variants, this script won't work, it used many syntax that not supported by them, the if syntax, the command substitution `$(...)`, the dot `.` like you said. Even the `read -r -p` won't.

**Comment by apjanke** [comment]
How about just removing the `source ~/.zshrc;` entirely? This script runs inside a noninteractive subshell, and under a shell which is not `zsh`. It doesn't seem to make a lot of sense to run a `zsh` startup file inside it.

I like the idea of adding a shebang line though. Not that it will do anything to the script's behavior as we use it, but it would serve as a nice bit of doco indicating that this file is to be executed as a script (instead of sourced) and indicating which shell it is to run under.

**Comment by Gnouc** [comment]
@apjanke Removing the source seems to be good. We should also update documentation and print some message to notice the users that they must restart their session.

If we're ok with that, I will update the changes.

**Comment by ncanceill** [comment]
I agree on removing the `source` statement altogether.

Thanks guys.

**Comment by Gnouc** [comment]
@ncanceill

**Comment by Gnouc** [comment]
What's state of this pull guys?

**Comment by mcornella** [comment]
Done. Thank you @Gnouc!

**Pull request merged**